sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -24,14 +24,70 @@
 #include "net/gnrc/udp.h"
 #include "net/sock/udp.h"
 #include "net/udp.h"
-#include "random.h"
 
 #include "gnrc_sock_internal.h"
 
 #ifdef MODULE_GNRC_SOCK_CHECK_REUSE
 static sock_udp_t *_udp_socks = NULL;
 #endif
 
+static uint16_t _dyn_port_next = 0;
+
+/* --------------------------------- helper --------------------------------- */
+
+/**
+ * @brief   Checks if a given UDP port is already used by another sock
+ */
+static bool _dyn_port_used(uint16_t port)
+{
+#ifdef MODULE_GNRC_SOCK_CHECK_REUSE
+    for (sock_udp_t *ptr = _udp_socks; ptr != NULL;
+         ptr = (sock_udp_t *)ptr->reg.next) {
+        bool spec_addr = false;
```

Comment:
`spec_addr`? Unspecified/specified might be the wrong terminology here, since it is more like `any` in sockets. As such me and @OlegHahm are a little bit confused about this whole section, since it may well be, that the sock is bound to any address, but the port is set. So it makes no sense to skip entries with all-zero addresses.